src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1190,30 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
```

Comment:
quick question:
can we use pattern matching (value instanceof Long len) here?  won't this create a problem in java8 ?

Reply:
As this don't get back ported to java 8 , so I guess it won't create any problem.